core/model/modx/moduser.class.php

```diff
@@ -787,7 +787,13 @@ public function removeLocks(array $options = array()) {
      * @param array $options
      * @return string The newly generated password
      */
-    public function generatePassword($length = 10,array $options = array()) {
+    public function generatePassword($length = null,array $options = array()) {
+        if ($length === null || $length === '') {
```

Comment:
After discussion with Jason.
Better to use such variant:
```
if ($length === null) {
    $length = $this->xpdo->getOption('password_generated_length', null, 10, true);
}
```
And return default value (10) to function signature.